### PR TITLE
[POAE7-2428]verify Primitive Type for Arrow Format

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
1. add PrimitiveTypeArrowTest & PrimitiveTypeQueryArrowTest
2. Move ArrowArrayBuilder::toString to CiderBatch::toStringForArrow
3. fix divided by 0 and memory leak bugs

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Unit Test

### CODE REVIEW

File: cider/tests/utils/QueryArrowDataGenerator.h [hunk 1]
```diff
@@ -234,7 +234,8 @@ class QueryArrowDataGenerator {
         offset_data.push_back(0);
         for (auto i = 0; i < row_num; ++i) {
           null_data[i] = Random::oneIn(null_chance, rng) ? true : false;
-          size_t str_len = rand() % (max_len + 1) + min_len;  // NOLINT
+          size_t str_len =
+              rand() % (max_len + 1) > 0 ? (max_len + 1) : 10 + min_len;  // NOLINT
           col_data += (random_string(str_len));
```
Comment (kd97) [inline cider/tests/utils/QueryArrowDataGenerator.h:239]: I think you mean data range is `rand() % (max_len - min_len  + 1) + min_len`?

Comment (kd97) [reply cider/tests/utils/QueryArrowDataGenerator.h:239]: And better to give a default str size.

File: cider/include/cider/batch/CiderBatch.h [hunk 2]
```diff
@@ -112,6 +112,13 @@ class CiderBatch {
 
   const void** getChildrenArrayPtr() const;
 
+  std::string toStringForArrow() const;
+
+  void printByTypeForArrow(std::stringstream& ss,
```
Comment (yma11) [inline cider/include/cider/batch/CiderBatch.h:117]: can this method moved out of `public`?

File: cider/tests/CiderArrowBatchTest.cpp [hunk 3]
```diff
@@ -757,33 +757,33 @@ TEST_F(CiderArrowBatchTest, ArrowEntryMoveTest) {
...
-        }
-      }
-      {
-        auto ptr = child2->asMutable<ScalarBatch<float>>()->getMutableRawData();
-        for (size_t i = 0; i < 100; ++i) {
-          CHECK_EQ(ptr[i], (float)i);
-        }
-      }
-    }
+    // auto [moved_schema, moved_array] = batch->move();
```
Comment (yma11) [inline cider/tests/CiderArrowBatchTest.cpp:760]: Maybe add a TODO here? seems we need to fix this in future.

File: cider/tests/functionality/PrimitiveTypeArrowTest.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+TEST_UNIT(PrimitiveTypeMixedTypeNotNullTest, mixedTypeNotNULLTest)
+
+TEST_UNIT(PrimitiveTypeMixedTypeNullTest, mixedTypeNULLTest)
+
+TEST_UNIT(PrimitiveTypeMixedTypeHalfNullTest, mixedTypeHalfNULLTest)
+
+int main(int argc, char** argv) {
+  testing::InitGoogleTest(&argc, argv);
+
```
Comment (yma11) [inline cider/tests/functionality/PrimitiveTypeArrowTest.cpp:195]: Add following lines at L195 for future easy debug:
`  logger::LogOptions log_options(argv[0]);
  log_options.parse_command_line(argc, argv);
  log_options.max_files_ = 0;  // stderr only by default
  logger::init(log_options);`

**Comment by yma11** [comment]
Please remove update about Thirdparty/Velox, there will be regular update by eco team so we don't do this. Another open is, do you copy **ALL** TCs from the original PrimitiveTypeTest and PrimitiveTypeQueryTest?

**Review by yma11 — COMMENTED** [review]
LGTM. @jikunshang, please also take a review.

**Review by yma11 — APPROVED** [review]

File: cider/include/cider/batch/CiderBatch.h [hunk 5]
```diff
@@ -145,6 +147,11 @@ class CiderBatch {
   bool ownership_{false};  // Whether need to release the tree of schema_ and array_.
   bool reallocate_{
       false};  // Whether permitted to (re-)allocate memory to buffers of array_.
+
+  void printByTypeForArrow(std::stringstream& ss,
```
Comment (winningsix) [inline cider/include/cider/batch/CiderBatch.h:151]: Minor hint: ```printByType``` directly since input ```ArrowArray``` already indicating Arrow type..

**Review by jikunshang — APPROVED** [review]

**Pull request merged**